### DISCUSSION TITLE
Weird formatting when using `<script setup>`

### DISCUSSION THREAD

**Discussion by rowild** [vuejs/language-tools#2735] [Q&A]
Hi!

Does anybody know what I have to do to red rid of this weird code highlightning?
(See closing `</script>` tag, then CSS rules.)

![Screenshot 2023-04-27 at 11 49 40](https://user-images.githubusercontent.com/213803/234827250-50510e8c-58c7-404b-aa15-516ce99f96f2.png)

Things are correct, when there is a semicolon after `defineProps`:

![Screenshot 2023-04-27 at 11 53 16](https://user-images.githubusercontent.com/213803/234827538-1d24e7b0-ffb3-4523-8331-b82516f69267.png)

but the prettier configuration is set to `semi: false`.

That rule seems to be contradicting with what Volar wants? Or is it a VS Code thingy?

**Comment by so1ve** [comment]
That is probably caused by the semicolon. Missing it may cause the generated virtual file to be broken. There are many exist issues reporting this, so you can track them or report a new one.

**Comment by rowild** [comment]
This is absolutely due to the semicolon, of course! But why does this happen with VS Code's standard theme? Installing other themes won't have any troubles with that code structure.

UPDATE:
I am wrong! Sorry! I just deactivated all extensions and activated only Volar. Then it works correctly. Now I try to find, which extension causes the culprit...

UPDATE 2:
...and of course I am not able to reproduce the error now. So thanks, @so1ve , for your feedback! Much appreciated! I will close this discussion now.

**Reply by so1ve** [reply]
> This is absolutely due to the semicolon, of course! But why does this happen with VS Code's standard theme? Installing other themes won't have any troubles with that code structure.
> 
> UPDATE: I am wrong! Sorry! I just deactivated all extensions and activated only Volar. Then it works correctly. Now I try to find, which extension causes the culprit...
> 
> UPDATE 2: ...and of course I am not able to reproduce the error now. So thanks, @so1ve , for your feedback! Much appreciated! I will close this discussion now.

Ah, try extension bisect.
